packages/gcf-utils/test/gcf-bootstrapper.ts

```diff
@@ -342,24 +384,206 @@ describe('GCFBootstrapper', () => {
-      req.headers['x-cloudtasks-taskname'] = '';
-      nockListInstallationRepos();
+    describe('per-repository cron', () => {
+      it('ensures that task is enqueued when called by scheduler for many repos', async () => {
+        await mockBootstrapper();
+        req.body = {
+          installation: {id: 1},
+        };
+        req.headers = {};
+        req.headers['x-github-event'] = 'schedule.repository';
+        req.headers['x-github-delivery'] = '123';
+        req.headers['x-cloudtasks-taskname'] = '';
+        const listInstallationRepoRequests = nockListInstallationRepos();
+
+        await handler(req, response);
+
+        sinon.assert.calledTwice(enqueueTask);
+        sinon.assert.notCalled(issueSpy);
+        sinon.assert.notCalled(repositoryCronSpy);
+        sinon.assert.notCalled(installationCronSpy);
+        sinon.assert.notCalled(globalCronSpy);
+        listInstallationRepoRequests.done();
+      });
 
-      await handler(req, response);
+      it('ensures that task is enqueued when called by scheduler for many installations', async () => {
+        await mockBootstrapper();
+        req.body = {};
+        req.headers = {};
+        req.headers['x-github-event'] = 'schedule.repository';
+        req.headers['x-github-delivery'] = '123';
+        req.headers['x-cloudtasks-taskname'] = '';
+        const listInstallationRequests = nockListInstallations();
+        const listInstallationRepoRequests = nockListInstallationRepos();
+
+        await handler(req, response);
+
+        sinon.assert.calledTwice(enqueueTask);
+        sinon.assert.notCalled(issueSpy);
+        sinon.assert.notCalled(repositoryCronSpy);
+        sinon.assert.notCalled(installationCronSpy);
+        sinon.assert.notCalled(globalCronSpy);
+        listInstallationRequests.done();
+        listInstallationRepoRequests.done();
+      });
+
+      it('handles the schedule.repository task', async () => {
+        await mockBootstrapper();
+        req.body = {
+          repo: 'test-owner/test-repo',
+          installation: {id: 1},
+          cron_type: 'repository',
+          cron_org: 'some-cron-org',
+        };
+        req.headers = {};
+        req.headers['x-github-event'] = 'schedule.repository';
+        req.headers['x-github-delivery'] = '123';
+        req.headers['x-cloudtasks-taskname'] = 'test-function';
+
+        await handler(req, response);
+
+        sinon.assert.notCalled(enqueueTask);
+        sinon.assert.notCalled(issueSpy);
+        sinon.assert.calledOnce(repositoryCronSpy);
+        sinon.assert.notCalled(installationCronSpy);
+        sinon.assert.notCalled(globalCronSpy);
+      });
+    });
+
+    describe('per-installation cron', () => {
+      it('ensures that task is enqueued when called by scheduler', async () => {
+        await mockBootstrapper();
+        req.body = {
+          cron_type: 'installation',
+        };
+        req.headers = {};
+        req.headers['x-github-event'] = 'schedule.repository';
```

Comment:
schedule.installation?

Reply:
This is actually hard-coded by serverless-scheduler-proxy: https://github.com/googleapis/repo-automation-bots/blob/e74ce012113a13631676cca20e1c4f148d19627e/serverless-scheduler-proxy/main.go#L76

When we queue the task, we change the name to the appropriate `schedule.type`.